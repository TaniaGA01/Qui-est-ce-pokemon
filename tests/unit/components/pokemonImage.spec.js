import { shallowMount } from "@vue/test-utils"
import PokemonPicture from "@/components/pokemonImage"

describe('PokemonPicture component', () => {

    test('Match with Snapshot', () => {

        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
        
    })

    test('Afficher image cachÃ©e et le pokemon 100', () => {

        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })

        const [img1, img2] = wrapper.findAll('img')

        expect( img1.exists() ).toBeTruthy()
        expect(img2).toBe(undefined)
        
        //pour s'assurer que l'image a la bonne classe :
        expect(img1.classes('hidden-pokemon')).toBeTruthy()
        
        // capturer l'image 100
        expect(img1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

        
    })
    
    test('Afficher Pokemon si showPokemon = true', () => {
        
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })

        const img1 = wrapper.find('img')

        expect( img1.exists() ).toBeTruthy()
        
        //pour s'assurer que l'image a la bonne classe :
        expect( img1.classes('hidden-pokemon')).toBeFalsy()
        expect(img1.classes('fade-in')).toBeTruthy()
        

    })
})