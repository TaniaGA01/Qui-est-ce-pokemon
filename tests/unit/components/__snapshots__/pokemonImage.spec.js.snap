// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component Match with Snapshot 1`] = `<div class="pokemon-container"><img class="hidden-pokemon fade-in" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon"></div>`;
